### PR TITLE
Use new Conversation.MaxMsgSummaries field

### PR DESCRIPTION
**Pull request by patrickxb**
This uses the new MaxMsgSummaries field instead of MaxMsgs.

### CODE REVIEW

File: shared/actions/chat.js [hunk 1]
```diff
@@ -396,13 +396,13 @@ function _conversationLocalToFinalized (convo: ?ChatTypes.ConversationLocal): Fi
 
 function _inboxToConversations (inbox: GetInboxLocalRes, author: ?string, following: {[key: string]: boolean}, metaData: MetaData): List<InboxState> {
   return List((inbox.conversationsUnverified || []).map(convoUnverified => {
-    const msgBoxed = convoUnverified.maxMsgs && convoUnverified.maxMsgs.length && convoUnverified.maxMsgs[0]
+    const msgMax = convoUnverified.maxMsgSummaries && convoUnverified.maxMsgSummaries.length && convoUnverified.maxMsgSummaries[0]
```
Comment (patrickxb) [inline shared/actions/chat.js]: @keybase/react-hackers please review my change in this function.

Comment (chrisnojima) [inline shared/actions/chat.js]: looking

Comment (chrisnojima) [inline shared/actions/chat.js]: This looks fine (logically and passes flow). I'd have to setup gregor locally to actually test this but i think this'll either work or not in your branch so i'm confident it'll be fine 👍

File: go/chat/inboxsource.go [hunk 2]
```diff
@@ -869,7 +872,7 @@ func (s *localizerPipeline) localizeConversation(ctx context.Context, uid gregor
 		conversationLocal.MaxMessages = msgs
 	} else {
 		conversationLocal.MaxMessages, err = s.G().ConvSource.GetMessagesWithRemotes(ctx,
```
Comment (mmaxim) [inline go/chat/inboxsource.go]: Could we just call `GetMessages` here? We could probably just delete `GetMessagesWithRemotes` after this change.

File: go/chat/convsource.go [hunk 3]
```diff
@@ -571,7 +578,7 @@ func (s *HybridConversationSource) GetMessages(ctx context.Context, convID chat1
 }
 
 func (s *HybridConversationSource) GetMessagesWithRemotes(ctx context.Context,
-	convID chat1.ConversationID, uid gregor1.UID, msgs []chat1.MessageBoxed,
+	convID chat1.ConversationID, uid gregor1.UID, msgs []chat1.MessageSummary,
```
Comment (mmaxim) [inline go/chat/convsource.go]: Yeah I think we should just nuke this function, it's not really doing anything more than `GetMessages` unless I am missing something.

Comment (patrickxb) [inline go/chat/convsource.go]: ok, done.

**Pull request merged**